test/e2e/readOnly.ts

```diff
@@ -0,0 +1,21 @@
+export const input = {
+  additionalProperties: false,
```

Comment:
What behavior would you expect if `readOnly` is used on the root schema? Seems like a good thing to support, though the spec is loosey goosey, as usual:

> An instance document that is marked as "readOnly for the entire document MAY be ignored if sent to the owning authority, or MAY result in an error, at the authority's discretion.

Reply:
Indeed, I did not thought about this one.
From a typescript point of view, having `ReadOnly<Root>` could be interesting.
However, I am a bit puzzled by the spec, from my understanding, it is not the modifications of the values of the document that should be ignored, but the whole document. So, I am not sure to see a matching typescript concept here.
What do you think?

Reply:
We can land this PR as-is (once it's updated to use a normalizer step), and track `readonly` for root schemas in a separate issue.  How does that sound?

Reply:
Sounds good to me